easybuild/easyblocks/p/petsc.py

```diff
@@ -79,6 +82,20 @@ def configure_step(self):
         Configure procedure is much more concise for older versions (< v3).
         """
         if LooseVersion(self.version) >= LooseVersion("3"):
+            # make the install dir first if we are doing a download install, then keep it for the rest of the way
+            if self.cfg["download_deps"] or self.cfg["download_deps_static"] or self.cfg["download_deps_shared"]:
+                self.log.info("Creating the installation directory before the configure.")
+                self.make_installdir()
+                self.cfg["keeppreviousinstall"] = True
+                deps = set( self.cfg["download_deps"]
+                            + self.cfg["download_deps_static"]
+                            + self.cfg["download_deps_shared"] )
```

Comment:
the wrapping makes this a bit messy, and you need to full list of deps in two places, so how about:

```python
        if LooseVersion(self.version) >= LooseVersion("3"):
            deps = self.cfg["download_deps"] + self.cfg["download_deps_static"] + self.cfg["download_deps_shared"]
            if deps:
                # make the install dir first if we are downloading dependencies, then keep it for the rest of the way
                ...
                for dep in set(deps):
                    ...
```